### PR TITLE
Feature/work with pin

### PR DESCRIPTION
**Pull request by fktt22**
Soundtrack of this PR: [Harlem Nocturne by Duke Ellington](https://www.youtube.com/watch?v=uIkekMoEQY4) PINs, security, checks... a throwback to noir films with hard private detectives!

### Motivation

Adding the usage of a PIN whenever MOB are sent.
Fixing a clash with Electron's own config.json file

### In this PR

Addition of a (hashed) PIN to the Electron store
Addition of hashing methods as utilities
Addition of a component to set the PIN in Settings

### Caveats

There's no way now to remove the PIN after having set one.
You have to rename `config.json` to `mobilecoind.json` 

### Future Work

Add PIN input and validation to sending MOB
Add unit testing

### Screen Shots

image
image

### Testing
```
Test Suites: 25 passed, 25 total
Tests:       107 passed, 107 total
Snapshots:   8 passed, 8 total
Time:        34.239 s
Ran all test suites.
Done in 37.92s.
```

### CODE REVIEW

**Comment by fktt22** [comment]
I had to modify the confirmation form, which was too high, in order to make space for the PIN entry. 

image

**Comment by fktt22** [comment]
On successful PIN entry, the SEND button is enabled.

image

**Comment by fktt22** [comment]
The PIN is input in the same way for a gift.

image

**Review by davidyhall — CHANGES_REQUESTED** [review]
Looks great! 3 small requests, the rest can be CBB.

1. The threshold should be inclusive. So a threshold of 1 MOB should trigger at 1 or great.
2. Let's require a the user's password to set/change the PIN
3. We're retiring `mobilecoind` (in the wallet!) so instead, lets name the file something like `mobilecoinConfig.json`

There are things that I think we can loop back on and improve later, but I think this gets the feature up and running!

**Comment by davidyhall** [comment]
Just a note: right now, the code hashes the PIN to keep the plain-text hidden. But it's a trivial operation to exhaust the options and discover the pin. The PIN isn't meant to function in the same capacity as the password, but there is a danger that the user has the same PIN for all of their devices/cards (highly not recommended!). So we don't want to compromise their personal PIN. We'll loop back and encrypt the PIN the same way we encrypt the entropy. And when the user unlocks the wallet with their password, we can keep the PIN in memory

**Comment by fktt22** [comment]
Require password to set/change PIN.
Show previous minimum for PIN.
Make limit inclusive (>=) instead of exclusive (>).
Minor fixes elsewhere.

File: app/layouts/DashboardLayout/TopBar/SyncStatus.tsx [hunk 1]
```diff
@@ -105,7 +105,7 @@ const SyncStatus: FC = () => {
           />
           {statusCode === SYNCING && (
             <CircularProgress
-              variant="static"
+              variant="determinate"
```
Comment (davidyhall) [inline app/layouts/DashboardLayout/TopBar/SyncStatus.tsx:108]: 🥇

File: app/utils/LocalStore.ts [hunk 2]
```diff
@@ -86,36 +88,33 @@ class LocalStore {
...
+  }
+
+  setMobilecoindDbPath(name: string): void {
+    this.store.set(schemaKeys.MOBILECOIND_DB_PATH, name);
   }
 
-  getName() {
-    return this.store.get(schemaKeys.NAME);
+  getName(): string {
+    return this.store.get(schemaKeys.NAME) as string;
```
Comment (davidyhall) [inline app/utils/LocalStore.ts:105]: TIL `as string` assertion.

**Review by davidyhall — APPROVED** [review]
LGTM!

**Pull request merged**